### PR TITLE
chore: Removes 1.20 deprecations

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Removes 1.20 deprecations

Link to any related issue(s): CLOUDP-274706

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by jeroenvervaeke — APPROVED** [review]

File: .changelog/2617.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+```release-note:breaking-change
```
Comment (AgustinBettati) [inline .changelog/2617.txt:1]: Should we add a mention in the migration guide as well?

Comment (EspenAlbert) [reply .changelog/2617.txt:1]: Added ✅ dee7b8938b5a4751e014745196ef89bd547b0dba

**Comment by maastha** [comment]
There's also some deprecations in federated database instance for Sept 2024 https://github.com/mongodb/terraform-provider-mongodbatlas/blob/02281f2ff9003a8ccf3011e96947fdbae5129e22/internal/service/federateddatabaseinstance/data_source_federated_database_instance.go#L232 
Should we remove these as well?

**Comment by EspenAlbert** [comment]
> There's also some deprecations in federated database instance for Sept 2024
> 
> https://github.com/mongodb/terraform-provider-mongodbatlas/blob/02281f2ff9003a8ccf3011e96947fdbae5129e22/internal/service/federateddatabaseinstance/data_source_federated_database_instance.go#L232
> 
> 
> Should we remove these as well?

I wasn't sure. Sept 2024, does it mean we remove it before or after? 😅

**Review by maastha — APPROVED** [review]
LGTM

File: internal/service/federateddatabaseinstance/data_source_federated_database_instance.go [hunk 2]
```diff
@@ -229,10 +228,6 @@ func schemaFederatedDatabaseInstanceStoresDataSource() *schema.Schema {
 					Type:     schema.TypeString,
 					Computed: true,
 				},
-				"cluster_id": {
```
Comment (lantoli) [inline internal/service/federateddatabaseinstance/data_source_federated_database_instance.go:232]: apart from removing the attribute from the schema, have you removed logic from the CRUD operations and tests?
same for all removed attributes

**Review by lantoli — APPROVED** [review]

File: internal/service/federateddatabaseinstance/data_source_federated_database_instance.go (hunk 2 above)
Comment (EspenAlbert) [reply internal/service/federateddatabaseinstance/data_source_federated_database_instance.go:232]: I was searching through and couldn't find any usage of `cluster_id` or `clusterId`

**Comment by EspenAlbert** [comment]
Bypassing docs approval to avoid blocking the release. Can do follow-up changes if necessary.
Failing tests are the same in dev but passes in QA

**Review by carriecwk — COMMENTED** [review]
LGTM

**Pull request merged**